daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md

```diff
@@ -138,45 +138,133 @@ The end-to-end flow is illustrated in the following diagram.
+The monitor pattern is recurring process that typically:
+
+1. Checks the status of a system
+1. Takes some action based on that status - e.g. send a notification
+1. Sleeps for some period of time
+1. Repeat
+
+The following diagram provides a rough illustration of this pattern.
+
+<img src="/images/workflow-overview/workflow-monitor-pattern.png" width=600 alt="Diagram showing how the monitor pattern works"/>
 
-1. The thread must first acquire the monitor. 
-1. Once the thread has acquired the monitor, it can access the shared resource.
-1. The thread then releases the monitor. 
+Depending on the business needs, there may be a single monitor or there may be multiple monitors, one for each business entity (for example, a stock). Furthermore, the amount of time to sleep may need to change, depending on the circumstances. These requirements make using cron-based scheduling systems impractical.
 
-This ensures that only one thread can access the shared resource at a time, preventing synchronization issues.
+Dapr Workflow supports this pattern natively by allowing you to implement _eternal workflows_. Rather than writing infinite while-loops ([which is an anti-pattern]({{< ref "workflow-features-concepts.md#infinite-loops-and-eternal-workflows" >}})), Dapr Workflow exposes a _continue-as-new_ API that workflow authors can use to restart a workflow function from the beginning with a new input.
 
-TODO: DIAGRAM?
+{{< tabs ".NET" >}}
+
+{{% codetab %}}
+
+```csharp
+public override async Task<object> RunAsync(WorkflowContext context, MyEntityState myEntityState)
+{
+    TimeSpan nextSleepInterval;
+
+    var status = await context.CallActivityAsync<string>("GetStatus");
+    if (status == "healthy")
+    {
+        myEntityState.IsHealthy = true;
+
+        // Check less frequently when in a healthy state
+        nextSleepInterval = TimeSpan.FromMinutes(60);
+    }
+    else
+    {
+        if (myEntityState.IsHealthy)
+        {
+            myEntityState.IsHealthy = false;
+            await context.CallActivityAsync("SendAlert", myEntityState);
+        }
+
+        // Check more frequently when in an unhealthy state
+        nextSleepInterval = TimeSpan.FromMinutes(5);
+    }
+
+    // Put the workflow to sleep until the determined time
+    await context.CreateTimer(nextSleepInterval);
+
+    // Restart from the beginning with the updated state
+    context.ContinueAsNew(myEntityState);
+    return null;
+}
+```
 
-In a few lines of code, you can create multiple monitors that observe arbitrary endpoints. The following code implements a basic monitor:
+Note that this example assumes you have a predefined `MyEntityState` class with a boolean `IsHealthy` property.
 
-TODO: CODE EXAMPLE
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+A workflow implementing the monitor pattern can loop forever or it can terminate itself gracefully by not calling _continue-as-new_.
```

Comment:
If this line only refers to the Monitor pattern, can we move up to the Monitor tab? Otherwise, it's fine as is!